app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
    * @return {Promise}
    */
   getAll() {
-    return this[p.net].fetchJSON(
-      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
-      // Get all recipes.
-      .then(services => services.filter(service =>
-        service.adapter !== 'thinkerbell-adapter'
-          && service.adapter !== 'webpush@link.mozilla.org'
-          && service.adapter !== 'console@link.mozilla.org'
-      ))
-      // Fetch their respective enabled status.
-      .then(recipes => {
-        const promises = recipes.map(recipe => {
-          const payload = [
-            { id: `${recipe.id}/get_enabled` }
-          ];
-
-          return this[p.net].fetchJSON(
-            `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/` +
-            'channels/get',
-            'PUT',
-            payload
-          );
+    let recipeServicesPromise = this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`,
+      'POST',
+      { getters: [{ kind: 'ThinkerbellRuleSource' }] }
+    );
+
+    // Mark getters and setters with more friendly names.
+    recipeServicesPromise = recipeServicesPromise.then(services => {
+      return services.map(service => {
+        const enabledGetterId = Object.keys(service.getters).find(getterId => {
+          return service.getters[getterId].kind === 'ThinkerbellRuleOn';
+        });
+        const sourceGetterId = Object.keys(service.getters).find(getterId => {
+          return service.getters[getterId].kind === 'ThinkerbellRuleSource';
+        });
+
+        const enabledSetterId = Object.keys(service.setters).find(setterId => {
+          return service.setters[setterId].kind === 'ThinkerbellRuleOn';
+        });
+        const removeSetterId = Object.keys(service.setters).find(setterId => {
+          return service.setters[setterId].kind === 'RemoveThinkerbellRule';
         });
 
-        return Promise.all(promises)
-          // Map all the recipes to a more user-friendly format.
-          .then(servicesEnabled => recipes.map((recipe, index) => ({
-            id: recipe.id,
-            label: recipe.id,
-            enabled: servicesEnabled[index][`${recipe.id}/get_enabled`]
-              && servicesEnabled[index][`${recipe.id}/get_enabled`]
-                .OnOff === 'On'
-          })));
+        return {
+          id: service.id,
+          getEnabled: enabledGetterId,
+          getSource: sourceGetterId,
+          setEnabled: enabledSetterId,
+          remove: removeSetterId,
+          status: null,
+          source: null
+        };
       });
-  }
+    });
 
-  getServicesWithGetters() {
```

Comment:
Refactoring is my only way to quickly get something done, sorry for the big changeset :/